lib/memoize.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const generateKey = args => args
+  .map(item =>
+    `${item.toString()}:${typeof item}`)
+  .join('|');
+
+const promisifyClosure = (fn, hasher) => (...args) => new Promise(
+  (resolve, reject) => {
+    fn(...args, (err, data) => {
+      const res = hasher(err, data) || data;
+      return (err ? reject(err) : resolve(res));
+    });
+  });
+
+const asyncMemoize = (fn, cacheSize, config = {}) => {
```

Comment:
Would it be better to pass the cacheSize in config? 
```suggestion
const asyncMemoize = (fn, config = {}) => {
```

Reply:
Yes, you're right